### PR TITLE
PWN-6887 - make empty state for history screens

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6887

## Description of Work

PWN-6887 - make empty state for history screens

## Screenshots (Optional)

Screenshot 1 | Screenshot 2 | Screenshot 3
:------------:|:-------------:|:------------:
[Android Emulator - Resizable_API_33:5554 2023-02-01 17-54-16] | [Android Emulator - Resizable_API_33:5554 2023-02-01 17-54-30] | [Paste Pic 3]

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/MainFragment.kt [hunk 1]
```diff
@@ -65,6 +71,9 @@ class MainFragment : BaseFragment(R.layout.fragment_main), MainTabsSwitcher, Cen
 
     private var lastSelectedItemId = R.id.homeItem
 
+    private val mainScreenActionButtonHandler: MainScreenActionButtonHandler?
+        get() = tabCachedFragments[R.id.homeItem] as? MainScreenActionButtonHandler
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/home/MainFragment.kt:75]: Why did you add a listener here, not directly in HomeFragment?

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/home/MainFragment.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/MainFragment.kt:75]: according to what we have there is no way no do it
we need to call function from historyFragment to homeFragment 
this was fast way

another way is to use DI or re write code and make calls to buy and receive in MainFragment 

or add same code in history fragment

also we can use deeplinks in app

but all this consumes time and I think must be done in separate task

what do you think ?

File: app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt [hunk 2]
```diff
@@ -99,6 +99,8 @@ class TokenHistoryFragment :
         errorStateLayout.buttonRetry.setOnClickListener { presenter.retryLoad() }
         historyRecyclerView.setupHistoryList()
         viewActionButtons.onButtonClicked = { onActionButtonClicked(it) }
+        emptyStateLayout.buttonBuy.setOnClickListener { onActionButtonClicked(ActionButton.BUY_BUTTON) }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt:102]: what's wrong with opening just Buy and Recieve fragments here?

File: app/src/main/res/drawable/ic_history_empty.xml [hunk 3]
```diff
@@ -0,0 +1,206 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (gslevinkov) [inline app/src/main/res/drawable/ic_history_empty.xml:1]: that's a large fcking svg, have you seen the warning from AS?

`Very long vector path (1432 characters), which is bad for performance. Considering reducing precision, removing minor details or rasterizing vector`

Comment (kabramovp2p) [reply app/src/main/res/drawable/ic_history_empty.xml:1]: on all big pictures with vectors we have this info message

File: app/src/main/res/layout/fragment_token_history.xml [hunk 4]
```diff
@@ -5,17 +5,20 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@color/backgroundPrimary"
+    android:background="@color/bg_smoke"
     android:orientation="vertical">
 
-    <androidx.appcompat.widget.Toolbar
+    <org.p2p.uikit.organisms.UiKitToolbar
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/res/layout/fragment_token_history.xml:11]: move to common layout for reuse

[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/res/layout/fragment_token_history.xml:11]: it doesn't solve the problem that we need to duplicate all the logic that fetches, filteres, builds the list on both screens

but thank you

**Review by gslevinkov — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]
